app/arena/battle.py

```diff
@@ -0,0 +1,22 @@
+from app.character.knight import Knight
+
+
+def battle(basic_knights: dict) -> dict:
+    knights_list = [Knight(**basic_knights[knight])
+                    for knight in basic_knights]
+
+    # gear_up() is to get weapon, armour, drink pots
+    [knight.gear_up() for knight in knights_list]
+
+    # fight with each other 1st vs 3rd, 2nd vs 4th
+    for i in range(len(knights_list) // 2):

```

Comment:
> what is i ?) use full/more descriptive names for vars

i is an index.

Reply:
i know that `i` is a common practice to name the index var
but if u have a 3-4 for loops with names like this i,x,y etc. it will really hard to read this 
what about knights_list_index ? or knight_index or smth similar to this ?